tests/regression/test_hypre_ams.py

```diff
@@ -121,18 +121,23 @@ def test_homogeneous_field_linear_convergence():
     constant_field = as_vector([-0.5*B0*(y - 0.5), 0.5*B0*(x - 0.5), 0])
 
     bc = DirichletBC(V, constant_field, (1, 2, 3, 4))
-
-    params = {'snes_type': 'ksponly',
-              'ksp_type': 'cg',
-              'ksp_max_it': '30',
-              'ksp_rtol': '1e-8',
-              'pc_type': 'python',
-              'pc_python_type': 'firedrake.HypreAMS',
-              'pc_hypre_ams_zero_beta_poisson': True
-              }
-
     A = Function(V)
     problem = LinearVariationalProblem(a, L, A, bcs=bc)
-    solver = LinearVariationalSolver(problem, solver_parameters=params)
-    solver.solve()
-    assert solver.snes.ksp.getIterationNumber() < 10
+
+    # test hypre options
+    for cycle_type in (1, 13):
+        expected = 9 if cycle_type == 1 else 6
+        params = {'snes_type': 'ksponly',
+                  'ksp_type': 'cg',
+                  'ksp_max_it': '30',
+                  'ksp_rtol': '1e-8',
+                  'pc_type': 'python',
+                  'pc_python_type': 'firedrake.HypreAMS',
+                  'pc_hypre_ams_zero_beta_poisson': True,
+                  'hypre_ams_pc_hypre_ams_cycle_type': cycle_type,
```

Comment:
Do we really want the `hypre_ams` prefix?

Reply:
We need a prefix of some kind, it could be shorter (e.g. `ams_`)?